cpp/src/arrow/filesystem/s3fs_benchmark.cc

```diff
@@ -0,0 +1,344 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <memory>
+#include <sstream>
+#include <utility>
+
+#include "benchmark/benchmark.h"
+
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/CreateBucketRequest.h>
+#include <aws/s3/model/HeadBucketRequest.h>
+#include <aws/s3/model/PutObjectRequest.h>
+
+#include "arrow/filesystem/s3_internal.h"
+#include "arrow/filesystem/s3_test_util.h"
+#include "arrow/filesystem/s3fs.h"
+#include "arrow/io/caching.h"
+#include "arrow/io/interfaces.h"
+#include "arrow/status.h"
+#include "arrow/table.h"
+#include "arrow/testing/gtest_util.h"
+#include "arrow/testing/random.h"
+
+#include "parquet/arrow/reader.h"
+#include "parquet/arrow/writer.h"
+#include "parquet/properties.h"
+
+namespace arrow {
+namespace fs {
+
+using ::arrow::fs::internal::ConnectRetryStrategy;
+using ::arrow::fs::internal::OutcomeToStatus;
+using ::arrow::fs::internal::ToAwsString;
+
+// Environment variables to configure the S3 test environment
+static const char* kEnvBucketName = "ARROW_TEST_S3_BUCKET";
+static const char* kEnvSkipSetup = "ARROW_TEST_S3_SKIP_SETUP";
+static const char* kEnvAwsRegion = "ARROW_TEST_S3_REGION";
+
+// Set up Minio and create the test bucket and files.
+class MinioFixture : public benchmark::Fixture {
+ public:
+  void SetUp(const ::benchmark::State& state) override {
+    ASSERT_OK(minio_.Start());
+
+    const char* region_str = std::getenv(kEnvAwsRegion);
+    if (region_str) {
+      region_ = region_str;
+      std::cerr << "Using region from environment: " << region_ << std::endl;
+    } else {
+      std::cerr << "Using default region" << std::endl;
+    }
+
+    const char* bucket_str = std::getenv(kEnvBucketName);
+    if (bucket_str) {
+      bucket_ = bucket_str;
+      std::cerr << "Using bucket from environment: " << bucket_ << std::endl;
+    } else {
+      bucket_ = "bucket";
+      std::cerr << "Using default bucket: " << bucket_ << std::endl;
+    }
+
+    client_config_.endpointOverride = ToAwsString(minio_.connect_string());
+    client_config_.scheme = Aws::Http::Scheme::HTTP;
+    if (!region_.empty()) {
+      client_config_.region = ToAwsString(region_);
+    }
+    client_config_.retryStrategy = std::make_shared<ConnectRetryStrategy>();
+    credentials_ = {ToAwsString(minio_.access_key()), ToAwsString(minio_.secret_key())};
+    bool use_virtual_addressing = false;
+    client_.reset(
+        new Aws::S3::S3Client(credentials_, client_config_,
+                              Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never,
+                              use_virtual_addressing));
+
+    MakeFileSystem();
+
+    const char* skip_str = std::getenv(kEnvSkipSetup);
+    const std::string skip = skip_str ? std::string(skip_str) : "";
+    if (skip == "true") {
```

Comment:
Shouldn't it be skipped if the variable has any non-empty value?

Reply:
Sounds good to me.